### PR TITLE
generate cyclicMeasures

### PR DESCRIPTION
**Pull request by anaim**

### CODE REVIEW

**Review by bennyvdhoogen — APPROVED** [review]

File: src/schema/measure/cyclic-measure.repository.ts [hunk 1]
```diff
@@ -63,6 +71,64 @@ export class CyclicMeasureRepository implements ICyclicMeasureRepository {
...
 
+	async generateCyclicMeasures(surveyId: string): Promise<CyclicMeasure[]> {
+		const cyclicMeasures: CyclicMeasure[] = [];
+		const units: Unit[] = await this.prisma.units.findMany({
+			where: { surveyId },
+		});
+
+		if (units) {
+			for (const unit of units) {
+				const unitCode: ObjectTypeUnitCode = await this.prisma.objectTypeUnitCodes.findFirst({
```
Comment (fortunato) [inline src/schema/measure/cyclic-measure.repository.ts:82]: I think ideally we don't mix concerns from different domains. This is a repository responsible for cyclicMeasures, not objectTypeUnitCodes.
I think this method/logic would be better situated in the command handler. The command handler can then have the necessary services and repositories (a kind of service really) injected to get the job done. This way we are not mixing concerns related to different domains/entities on the service/repository level.

File: src/schema/measure/utils/cast-cyclic-measure-type.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { CyclicMeasureTypes } from '../types/cyclic-measure';
+
+export const castCyclicMeasureType = (maintenanceType: string) => {
```
Comment (fortunato) [inline src/schema/measure/utils/cast-cyclic-measure-type.ts:3]: Neat. Is it worth making it explicit that the return type is an enum or a string?

**Review by fortunato — CHANGES_REQUESTED** [review]

File: src/schema/measure/cyclic-measure.repository.ts (hunk 1 above)
Comment (anaim) [reply src/schema/measure/cyclic-measure.repository.ts]: Done

File: src/schema/measure/utils/cast-cyclic-measure-type.ts (hunk 2 above)
Comment (anaim) [reply src/schema/measure/utils/cast-cyclic-measure-type.ts]: Done

File: src/schema/measure/dto/create-cyclic-measure.input.ts [hunk 3]
```diff
@@ -5,8 +5,8 @@ import { BaseCyclicMeasureInput } from './base-cyclic-measure.input';
 
 @InputType()
 export class CreateCyclicMeasureInput extends BaseCyclicMeasureInput {
-	@Field()
-	public remarks: string;
+	// @Field()
```
Comment (fortunato) [inline src/schema/measure/dto/create-cyclic-measure.input.ts:8]: Can this be removed?

File: src/schema/measure/commands/generate-cyclic-measures.handler.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+								planYear: 0,
+								finalPlanYear: 0,
+								costSurcharge: 0,
+								remarks: '',
+								failureModeId: '',
+								defectId: '',
+							};
+
+							cyclicMeasures.push(
+								(await this.cyclicMeasureRepository.createCyclicMeasure(
```
Comment (fortunato) [inline src/schema/measure/commands/generate-cyclic-measures.handler.ts:67]: Can we use a factory to transform a Prisma result to a CyclicMeasure GQL model?

File: src/schema/measure/commands/generate-cyclic-measures.handler.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+							const data: DomainCyclicMeasure = {
+								...existingCyclicMeasure,
+								cycle: defaultMaintenanceMeasure.cycle,
+								unitPrice: defaultMaintenanceMeasure.unitPrice,
+								quantityUnitOfMeasurement: defaultMaintenanceMeasure.quantityUnitOfMeasurement,
+								maintenanceType: castCyclicMeasureType(defaultMaintenanceMeasure.maintenanceType),
+							};
+
+							cyclicMeasures.push(
+								(await this.cyclicMeasureRepository.updateCyclicMeasure(
```
Comment (fortunato) [inline src/schema/measure/commands/generate-cyclic-measures.handler.ts:81]: Can we use a factory to transform a Prisma result to a CyclicMeasure GQL model?

**Review by fortunato — CHANGES_REQUESTED** [review]

**Review by fortunato — APPROVED** [review]

File: src/schema/measure/commands/generate-cyclic-measures.handler.ts (hunk 4 above)
Comment (anaim) [reply src/schema/measure/commands/generate-cyclic-measures.handler.ts]: Done

File: src/schema/measure/commands/generate-cyclic-measures.handler.ts (hunk 5 above)
Comment (anaim) [reply src/schema/measure/commands/generate-cyclic-measures.handler.ts]: Done

**Pull request merged**